### DISCUSSION TITLE
U-turn instructions

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#328] [Q&A] [answered]
Some voice prompts are suboptimal.
e.g.: before some hairpin bends comes: "Wenden" That means: Turn Around!...not optimal.
Better maybe :"180grad Kurve" or "scharfe Kurve"....

_Originally posted by @Leeb64 in https://github.com/devemux86/cruiser/discussions/327#discussion-5665554._

**Comment by devemux86** [comment]
@Leeb64 thanks for the suggestion.

We can all discuss what is the best way to inform for u-turns.

The English texts are:
- make a U-turn
- make a U-turn left
- make a U-turn right

**Reply by bios4** [reply]
Simple hairpin turns (e.g. on alpine passes) as mentioned by Leeb64 should not generate any voice indications at all in my opinion, as long as you simply have to follow the course of the road!??

If, on the other hand, a route requires a "real" u-turn (e.g. in cities where one has to take into account one-way-streets), then of course a voice prompt should inform properly...

**Comment by WolfgangZ42** [comment]
> before **_some_** hairpin bends comes: "Wenden" 

Must be really only for "some" hairpin curves - I drive a lot on alpine road and can't remember of "u-turns" ... 
Here two passes for testing with simulation: [Passes.zip](https://github.com/devemux86/cruiser/files/12709653/Passes.zip) - not any "u-turn". I assume, that in some rarely cases the definition in OSM is wrong and should be optimized.

**Comment by devemux86** [comment, marked as answer]
_Reminder: the turn instructions are generated by each routing engine._

Cruiser just receives the turn instructions and uses them in navigation.

---

There is already an idea for Cruiser to post-process the turn instructions and remove some unnecessary "Continue":

- https://github.com/devemux86/cruiser/discussions/307

**Comment by DolomitesTom** [comment]
"Spitzkehre" links/rechts could be the german word

**Reply by bios4** [reply]
That's basically correct - but why indicate it per voice prompt at all???

If i don't have to actually TURN (meaning any action which requires me to use my indicators) it is just the course of the road, right?
So no need to be informed about that...

Imagine riding the Stelvio pass and being informed every 200-300 meters about a hairpin. 😄
![stelvio](https://github.com/devemux86/cruiser/assets/145460147/2509f90a-2bcc-4291-a781-77791ab2a242)

**Reply by devemux86** [reply]
> it is just the course of the road

This depends on how each router generates the turn instructions.

Is it about BRouter (can use [BRouter website](https://brouter.de/brouter-web/) for route examples) or another router?
